modules/creating_a_blueprint.md

```diff
@@ -145,6 +145,34 @@ There are multiple ways to customize a phase from JHipster.
     }
 ```
 
+4) Re-order the phase steps, this is when the blueprint gets the phase from JHipster and generates a new step sequence.
+
+```javascript
+    // Run the blueprint steps before any parent steps
+    get initializing() {
+        const phaseFromJHipster = 
+        return {
+            myCustomInitPhaseStep() {
+                // Do all your stuff here
+            },
+            ...super._initializing();
+        }
+    }
+    // Fine control of the steps order
+    get configuring() {
+        const phaseFromJHipster = super._configuring();
+        return {
+            insight: phaseFromJHipster.insight,
```

Comment:
The problem with this approach is that if generator-jhipster adds a new step, the blueprint will fail and it will be difficult the find the problem.
Maybe add a verification if the result has every key from jhipsterStep.

Reply:
Something like 
```
for (step in phaseFromJHipster) {
  customPhase.hasOwnProperty(step)
}
```
In one hand, I undertand the additional verification but it the other it looks tricky.
Overall re-ordering steps implies a strong coupling between the upstream generator and the blueprint.
Maybe just adding a warning would be enough:
> While finely control the phase steps gives you a great flexibility, it's  error prone. If generator-jhipster adds a new step, the blueprint will fail and it will be difficult the find the problem. Use it at your own risk.

WDYT?

Reply:
Sure.

Reply:
I reconsider my position. I don't think a fine control on steps order is something to promote. As you said @mshima it's risky and hard to troubleshoot. I think a basic decoration (pre/post phase) should do the job.